### DISCUSSION TITLE
Can't uninstall Blackhole 2ch

### DISCUSSION THREAD

**Discussion by dudeon41** [ExistentialAudio/BlackHole#721] [General]
### Please Confirm

- [X] I have read the **[FAQ](https://github.com/ExistentialAudio/BlackHole#faq) and [Wiki](https://github.com/ExistentialAudio/BlackHole/wiki)** where most common issues can be resolved
- [X] I have searched **[Discussions](https://github.com/ExistentialAudio/BlackHole/discussions)** to see if the same question has already been asked
- [X] This is a **bug** and not a question about audio routing or configuration, which should be posted in [Discussions](https://github.com/ExistentialAudio/BlackHole/discussions)

### macOS Version

macOS 13 Ventura

### BlackHole Build(s) Affected

- [X] 2 channel
- [ ] 16 channel
- [ ] 64 channel
- [ ] other/custom build

### Describe the bug

Hi, I'm not sure if this belongs here or in the Discussions section, but I'm trying to uninstall Blackhole 2ch, but I must've messed something up somewhere previously, because neither the uninstaller app nor the manual instructions work to uninstall it.

When trying to install the uninstaller app I get the error "The installation failed. The installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance."

When trying to uninstall manually there is no HAL folder at all under /Library/Audio/Plug-Ins/HAL and the sudo commands listed on this page: https://github.com/ExistentialAudio/BlackHole/wiki/Uninstallation do not seem to do anything.

Blackhole 2ch is still listed under Audio MIDI setup, Settings > Sound > Output & Input as well as in a couple other apps I use like Ableton and eqMac.

I tried reinstalling Blackhole as well, but get the same error as using the uninstaller app, which is "The installation failed. The installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance."

Any help would be greatly appreciated. Thanks!

### Reproduction Steps

Listed above in more detail, but trying to uninstall/reinstall the app or find the current program files.

### Expected Behavior

To uninstall the program or remove any existing corrupted files.

**Comment by devinroth** [comment]
Check to make sure "Installer" has File and Folder permission and Full Disk Access in System Settings->Privacy and Security.

**Reply by tconroy** [reply]
Hi @devinroth, I don't actually see "Installer" Listed in the `File and Folder` or `Full Disk Access` lists. How do we add it?

![CleanShot 2024-04-21 at 15 54 33](https://github.com/ExistentialAudio/BlackHole/assets/1609336/99706b68-f59c-4eb4-affc-cdc42a9bfb5c)

**Comment by vw110193** [comment]
Also getting this issue where there is no folder and I can't uninstall it

**Reply by hussein-javani** [reply]
I tried this and worked for me:
1. `cd` to the root of the computer ( for example Macintosh HD). you can use `cd ~/../..` command when you open the terminal
2. use `cd Library/Audio/Plug-Ins` to navigate to the main computer audio plugins forlder instead of the user specific one
3. use `rm -R HAL` to remove the whole "HAL" folder instead of removing the driver file since I could not find any drivers inside the folder. ( the file system does not show the blackhole drivers and I dont know why )
4. use `sodu killall -9 coreaudio` to reset the audo system.

**Comment by devinroth** [comment]
Screenshots please.

And have you tried restarting your computer?

Devin

**Comment by tconroy** [comment]
Also running into this issue...

1. there is no driver related to blackhole in `/Library/Audio/Plug-Ins/HAL`:
```
❯ ls -lah
Permissions Size User Group Date Modified    Name
drwxr-xr-x     - root wheel 2022-11-28 16:31  MSTeamsAudioDevice.driver/
/Library/Audio/Plug-Ins/HAL ❯                                                                                                                 03:58:45 PM
```

2. both blackhole drivers still show in Audio/Midi devices, apps, etc:
![CleanShot 2024-04-21 at 15 59 43](https://github.com/ExistentialAudio/BlackHole/assets/1609336/b5fbe17e-4855-4fdd-bacb-78c7697c3cc0)


3. when trying to run the uninstaller, it errors:
![CleanShot 2024-04-21 at 16 00 17](https://github.com/ExistentialAudio/BlackHole/assets/1609336/b76b6305-2e88-4901-8a40-911b8a7067cd)

**Comment by devinroth** [comment]
Try just restarting your computer.

Devin

**Comment by tech-n-code** [comment]
Use the manual uninstall method `rm -R /Library/Audio/Plug-Ins/HAL/BlackHoleXch.driver` (replacing 'x' with version).
Followed by `sudo killall coreaudiod` instead of `sudo launchctl kickstart -kp system/com.apple.audio.coreaudiod` to restart the service. Source: https://forums.developer.apple.com/forums/thread/748228

I installed it with Homebrew and uninstalling it the same way would not remove it from `brew list` so I also had to delete its' folder from `/opt/homebrew/Caskroom`

**Comment by Texula** [comment]
Having the same issue now, even downloading the latest version...

**Reply by Texula** [reply]
Has anybody found a fix?

**Comment by devinroth** [comment]
Yeah. It’s already in the discussions.

Devin

**Comment by kudatama** [comment]
Check if you have the right uninstaller, you can download the uninstaller here: existential.audio/blackhole/support. If you installed BlackHole 2ch, remove it with uninstaller 2ch.

I had the same problem of having uninstallation failure. I thought it was a well-hidden malware, and was gonna stuck there forever. I made a mistake of uninstalling the wrong version. So check if you got the wrong uninstaller; if so, get the right one.

**Reply by ihafkenschiel** [reply]
This worked for me. Thanks!

**Reply by PgmJun** [reply]
This worked for me too.
Thank you!

**Comment by davidrgmcb** [comment]
Running into this issue as well, attempted manual removal, it says that it's gone, used the correct 2ch uninstaller, fails every time. Despite all this, it still shows up as an audio source.

**Reply by Texula** [reply]
> Running into this issue as well, attempted manual removal, it says that it's gone, used the correct 2ch uninstaller, fails every time. Despite all this, it still shows up as an audio source.

Just reboot you computer! It solved the issue for me.

**Reply by davidrgmcb** [reply]
Tried it, sadly no luck.

**Comment by devinroth** [comment]
Rogue Ameoba’s ACE driver has been messing with CoreAudio. If you have that installed remove that and see if it fixes the issue.

Devin

**Reply by davidrgmcb** [reply]
Removing the ACE driver did allow me to remove blackhole, thank you so much.

**Comment by takechanman1228** [comment]
I ran into the exact same issue where the BlackHole driver folder was missing and the uninstaller app kept failing. 
My workaround was to **re-install BlackHole** (which recreated the /Library/Audio/Plug-Ins/HAL/BlackHoleXch.driver folder). After that, I could manually remove the driver using the sudo rm -rf /Library/Audio/Plug-Ins/HAL/BlackHoleXch.driver command.

I believe the uninstaller app may have failed, causing the folder to disappear. Hope this helps anyone else stuck with the same problem!

**Reply by andiwijayaid** [reply]
It works, thanks!

**Comment by Hesi-Re** [comment]
Running into this issue as well
